core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -472,16 +475,45 @@ private[spark] class MapOutputTrackerMaster(
     shuffleStatuses.get(shuffleId).map(_.findMissingPartitions())
   }
 
+  /**
+   * Try to equally divide Range(0, num) to divisor slices
+   */
+  def equallyDivide(num: Int, divisor: Int): Iterator[Seq[Int]] = {
+    assert(divisor > 0, "Divisor should be positive")
+    val (each, remain) = (num / divisor, num % divisor)
+    val (smaller, bigger) = (0 until num).splitAt((divisor-remain) * each)
```

Comment:
can you add some comment to describe the algorithm? I'd expect something like:
```
to equally divide n elements to m buckets
each bucket should have n/m elements
for the remaining n%m elements
pick the first n/m buckets and add one more element
```

Reply:
Sure : )

Reply:
my proposal
```
def equallyDivide(numElements: Int, numBuckets: Int) {
  val elementsPerBucket = numElements / numBuckets
  val remaining = numElements % numBuckets
  if (remaining == 0) {
    0.until(num).grouped(elementsPerBucket)
  } else {
    val splitPoint = (elementsPerBucket + 1) * remaining
    0.to(splitPoint).grouped(elementsPerBucket + 1) ++ (splitPoint + 1).until(numElements).grouped(elementsPerBucket)
  }
}
```